File: Cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float dollars;
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;
    int coins = 0;

    do
    {
        dollars = get_float("Change owed: "); // prompt dollars
    }
    while (dollars < 0);

    int cents = round(dollars * 100); // convert to change

    while (cents >= quarter) // quarters required
    {
        cents -= quarter;
        coins++;
    }

    while (cents >= dime) // dimes required
    {
        cents -= dime;
        coins++;
    }

    while (cents >= nickel) // nickels required
    {
        cents -= nickel;
        coins++;
    }

    while (cents >= penny) // pennies required
    {
        cents -= penny;
        coins++;
    }
    while (true) // print amount of coins
    {
        printf("%i\n", coins);
        break;
    }
}

File: cash.py
from cs50 import get_float

# assign variables to US coin equivalent and create variable for coins to increment
quarter = 25
dime = 10
nickel = 5
penny = 1
coins = 0

dollars = get_float("Change owed: ")  # prompt for input
while dollars < 0:
    dollars = get_float("Change owed: ")  # reprompt if entry invalid
cents = round(dollars * 100)  # round dollars to 100

while cents >= quarter:
    cents -= quarter
    coins += 1

while cents >= dime:
    cents -= dime
    coins += 1

while cents >= nickel:
    cents -= nickel
    coins += 1

while cents >= penny:
    cents -= penny
    coins += 1

print(f"{coins}")

